src/lib/routing-tables.js

```diff
@@ -235,103 +232,52 @@ class RoutingTables {
-      finalLedger: finalLedger,
-      finalAmount: finalAmount, // CAUTION: this will be used as destinationAmount
-      minMessageWindow: nextHop.bestRoute.minMessageWindow,
-      liquidityCurve: nextHop.bestRoute.curve && nextHop.bestRoute.curve.getPoints(),
-      additionalInfo: isFinal ? nextHop.bestRoute.additionalInfo : undefined
-    })
+  getLocalRoute (ledgerA, ledgerB) {
+    return this._getRoute(ledgerA, ledgerB, PAIR)
   }
 
   /**
-   * @param {IlpAddress} sourceAddress
-   * @param {IlpAddress} finalAddress
+   * @param {IlpAddress} source
+   * @param {IlpAddress} destination
    * @param {String} sourceAmount
    * @returns {Object}
    */
-  findBestHopForSourceAmount (sourceAddress, finalAddress, sourceAmount) {
-    const nextHop = this._findBestHopForSourceAmount(sourceAddress, finalAddress, +sourceAmount)
-    if (!nextHop) return
-    const sourceLedger = nextHop.bestRoute.sourceLedger
-    const finalLedger = nextHop.bestRoute.destinationLedger
-    const nextLedger = nextHop.bestRoute.nextLedger
-    const routeFromAToB = this._getLocalPairRoute(sourceLedger, nextLedger)
-    const isFinal = nextLedger === finalLedger
-    return omitUndefined({
-      isFinal: isFinal,
-      isLocal: nextHop.bestRoute.isLocal,
-      sourceLedger: sourceLedger,
-      sourceAmount: sourceAmount,
-      destinationLedger: nextLedger,
-      destinationAmount: routeFromAToB.amountAt(+sourceAmount).toString(),
-      destinationCreditAccount: isFinal ? null : nextHop.bestHop,
-      finalLedger: finalLedger,
-      finalAmount: nextHop.bestValue,
-      minMessageWindow: nextHop.bestRoute.minMessageWindow,
-      liquidityCurve: nextHop.bestRoute.curve && nextHop.bestRoute.curve.getPoints(),
-      additionalInfo: isFinal ? nextHop.bestRoute.additionalInfo : undefined
-    })
-  }
-
-  _findBestHopForSourceAmount (source, destination, amount) {
-    debug('searching best hop from %s to %s for %s (by src amount)', source, destination, amount)
+  findBestHopForSourceAmount (source, destination, sourceAmount) {
+    debug('searching best hop from %s to %s for %s (by src amount)', source, destination, sourceAmount)
     const table = this.sources.resolve(source)
     if (!table) {
       debug('source %s is not in known sources: %s',
         source, Object.keys(this.sources.prefixes))
       return undefined
     }
     return this._rewriteLocalHop(
-      table.findBestHopForSourceAmount(destination, amount))
+      table.findBestHopForSourceAmount(destination, sourceAmount))
   }
 
-  _findBestHopForDestinationAmount (source, destination, amount) {
-    debug('searching best hop from %s to %s for %s (by dst amount)', source, destination, amount)
+  /**
+   * Find the best intermediate ledger (`nextLedger`) to use after `sourceLedger` on
+   * the way to `finalLedger`.
+   * This connector must have `[sourceLedger, nextLedger]` as a pair.
+   *
+   * @param {IlpAddress} source
+   * @param {IlpAddress} destination
```

Comment:
Let's remove all use of the term 'destination' because it's never clear to me in which places it is used to mean 'next' and in which places it is used to mean 'final'. It's also not clear from this whether `source` is a ledger or an account, or whether both are allowed